File: app/src/main/java/org/evcs/android/ui/view/profile/ListButtonView.java
package org.evcs.android.ui.view.profile;

import android.content.Context;
import android.content.res.TypedArray;
import androidx.annotation.StringRes;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import org.evcs.android.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ListButtonView extends LinearLayout {

    @BindView(R.id.view_list_button_text) TextView mText;
    @BindView(R.id.view_list_button_bottom_line) View mBottonLine;

    public ListButtonView(Context context) {
        super(context);
    }

    public ListButtonView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public ListButtonView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        View view = inflate(context, R.layout.view_list_button, this);
        ButterKnife.bind(this, view);

        TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.ListButtonView);
        String text = typedArray.getString(R.styleable.ListButtonView_text);
        boolean bottomDiv = typedArray.getBoolean(R.styleable.ListButtonView_bottomDivider, true);
        typedArray.recycle();

        mText.setText(text);
        mBottonLine.setVisibility(bottomDiv ? VISIBLE : INVISIBLE);
        setClickable(true);
    }

    public void setText(String text) {
        mText.setText(text);
    }

    public void setText(@StringRes int resId) {
        mText.setText(resId);
    }

}

File: app/src/main/java/org/evcs/android/ui/view/profile/ListButtonView.kt
package org.evcs.android.ui.view.profile

import android.content.Context
import android.text.TextUtils
import android.widget.LinearLayout
import android.widget.TextView
import android.view.LayoutInflater
import android.util.AttributeSet
import org.evcs.android.R
import androidx.annotation.StringRes
import org.evcs.android.databinding.ViewListButtonBinding

class ListButtonView : LinearLayout {

    private lateinit var mText: TextView

    constructor(context: Context?) : super(context) {}

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) :
            super(context, attrs, defStyleAttr) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        val mBinding = ViewListButtonBinding.inflate(LayoutInflater.from(context), this, true)
        mText = mBinding.viewListButtonText
        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.ListButtonView)
        val text = typedArray.getString(R.styleable.ListButtonView_text)
        val label = typedArray.getString(R.styleable.ListButtonView_lbv_label)
        val bottomDiv = typedArray.getBoolean(R.styleable.ListButtonView_bottomDivider, true)
        val showButton = typedArray.getBoolean(R.styleable.ListButtonView_showButton, true)
        typedArray.recycle()
        mText.text = text
        mBinding.viewListButtonLabel.text = label
        mBinding.viewListButtonLabel.visibility = if (TextUtils.isEmpty(label)) GONE else VISIBLE
        mBinding.viewListButtonBottomLine.visibility = if (bottomDiv) VISIBLE else INVISIBLE
        mBinding.viewListButtonChevron.visibility = if (showButton) VISIBLE else GONE
        isClickable = showButton
    }

    fun setText(text: String?) {
        mText.text = text
    }

    fun setText(@StringRes resId: Int) {
        mText.setText(resId)
    }
}